utils/release_tracking.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import re
+import os
+import subprocess
+import sys
+
+
+def run_cargo_build(path):
+    print(f"cargo build, path: {path}")
+    command = f'cargo build'
+    subprocess.check_output(command.split(), cwd=path)
+
+
+def clone_repo(clone_dir, repo_name, branch):
+    git_repo = f"https://github.com/parallaxsecond/{repo_name}.git"
+    future_repo_dir = os.path.join(clone_dir, repo_name)
+    if not os.path.isdir(future_repo_dir):
+        command = f"git clone {git_repo} -b {branch} {future_repo_dir}"
+        subprocess.check_output(command.split())
+        command = f"git submodule update --init"
+        subprocess.check_output(command.split(), cwd=future_repo_dir)
+
+
+def git_toml_deps(toml_path, updatable_deps, deps_repos):
+    lines = None
+    with open(toml_path, 'r') as f:
+        lines = f.readlines()
+
+    for i in range(len(lines)):
+        line = lines[i]
+        for dep in updatable_deps.keys():
+            if line.startswith(dep + " ="):
+                # All occurences
+                line = line.replace("'", '"')
+                if "{" not in line:
+                    # First occurence
+                    line = line.replace('"', '{ version = "', 1)
+                    line = line.rstrip() + ' }\n'
+
+                if 'path' not in line:
+                    line = re.sub(r'version = "[0-9\.]+"', f'path = "{deps_repos[dep]}"', line)
+                lines[i] = line
```

Comment:
Modifying an list while we are iterating through it is pretty risky. 

    output_lines = []

    for line in lines():
        output_lines.append(lines)

Reply:
Done